barretenberg/cpp/src/barretenberg/stdlib/primitives/safe_uint/safe_uint.cpp

```diff
@@ -135,6 +135,7 @@ safe_uint_t<Builder> safe_uint_t<Builder>::divide(
 
     this->assert_equal(int_val, "divide method quotient and/or remainder incorrect");
 
+    quotient.set_origin_tag(OriginTag(get_origin_tag(), other.get_origin_tag()));
```

Comment:
why do you only set the origin tag for divide and slice? if it's an implementation detail related to the difference in how results are constructed it would be nice to be documented

Reply:
I guess tags get set properly for the other operations via the logic in the underlying field_t

Reply:
Yes, we only need to do it manually for elements that are created independently (without field_t operations)